crates/router/src/core/webhooks.rs

```diff
@@ -459,9 +459,21 @@ async fn trigger_webhook_to_merchant<W: api::OutgoingWebhookType>(
                 .change_context(errors::WebhooksFlowError::CallToMerchantFailed)?;
         }
         Ok(res) => {
+            let update_event = storage::EventUpdate {
+                event_type: None,
+                event_class: None,
+                is_webhook_notified: Some(true),
+                intent_reference_id: None,
+                primary_object_id: None,
+                primary_object_type: None,
```

Comment:
If the fields which are passed as `None` are never to be updated, would be better to remove them from the update object.